docs/source/tutorial_metrics.md

```diff
@@ -0,0 +1,372 @@
+## Introduction and Standard Metrics
+
+ParlAI contains a number of built-in metrics that are automatically computed when
+we train and evaluate models. Some of these metrics are _text generation_ metrics,
+which happen any time we generate a text: this includes F1, BLEU and Accuracy.
+
+For example, let's try a Fixed Response model, which always returns a given fixed
+response, and evaluate on the DailyDialog dataset:
+
+```
+$ parlai eval_model -m fixed_response -t dailydialog --fixed-response "how may i help you ?"
+... after a while ...
+14:41:40 | Evaluating task dailydialog using datatype valid.
+14:41:40 | creating task(s): dailydialog
+14:41:41 | Finished evaluating tasks ['dailydialog'] using datatype valid
+    accuracy  bleu-4  exs    f1
+    .0001239 .002617 8069 .1163
+```
+
+We see that we got 0.01239% accuracy, 0.26% BLEU-4 score, and 11.63% F1 across
+8069 examples. What do those metrics means?
+
+- Accuracy: this is perfect, exact, matching of the response, averaged across
+  all examples in the dataset
+- BLEU-4: this is the [BLEU score](https://en.wikipedia.org/wiki/BLEU) between
+  the predicted response and the reference response. It is measured on
+  tokenized text, and uses NLTK to compute it.
+- F1: This is the [Unigram](https://en.wikipedia.org/wiki/N-gram) F1 overlap
+  between your text and the reference response.
+- exs: the number of examples we have evaluated
+
+If you don't see the BLEU-4 score, you may need to install NLTK with
+`pip install nltk`.
+
+We can also measure ROUGE. Note that we need to `pip install py-rouge` for this
+functionality:
+
+```
+$ parlai eval_model -m fixed_response -t dailydialog --fixed-response "how may i help you ?" --metrics rouge
+14:47:24 | creating task(s): dailydialog
+14:47:31 | Finished evaluating tasks ['dailydialog'] using datatype valid
+    accuracy  exs    f1  rouge_1  rouge_2  rouge_L
+    .0001239 8069 .1163   .09887  .007285   .09525
+```
+
+### Agent-specific metrics
+
+Some agents include their own metrics that are computed for them. For example,
+generative models automatically compute `ppl`
+([perplexity](https://en.wikipedia.org/wiki/Perplexity)) and `token_acc`, both
+which measure the generative model's ability to predict indivdual tokens.  As
+an example, let's evaluate the [BlenderBot](https://parl.ai/projects/recipes/)
+90M model on DailyDialog:
+
+```
+$ parlai eval_model --task dailydialog -mf zoo:blender/blender_90M/model -bs 32
+...
+14:54:14 | Evaluating task dailydialog using datatype valid.
+14:54:14 | creating task(s): dailydialog
+...
+15:26:19 | Finished evaluating tasks ['dailydialog'] using datatype valid
+    accuracy  bleu-4  ctpb  ctps  exps  exs    f1  gpu_mem  loss      lr  ltpb  ltps   ppl  token_acc  total_train_updates   tpb   tps
+           0 .002097 14202 442.5 6.446 8069 .1345    .0384 2.979 7.5e-06  3242   101 19.67      .4133               339012 17445 543.5
+```
+
+Here we see a number of extra metrics, each of which we explain below:
```

Comment:
It seems to me that some of these are metrics related to the performance of the model/algorithm. And the rest are related to complexity of it and how that is handled. If that's the case, would you mind separating the two?